citations-impl/impl/src/java/org/sakaiproject/citation/impl/BaseCitationService.java

```diff
@@ -3226,12 +3232,16 @@ protected void set(BasicCitationCollection other, boolean isTemporary)
 						BasicCitation oldCitation = (BasicCitation) collection.getCitation(citationCollectionOrder.getCitationid());
 						BasicCitation newCitation = new BasicCitation();
 						newCitation.copy(oldCitation);
+						newCitation.m_id = oldCitation.m_id;
 						newCitation.m_temporary = isTemporary;
 						this.saveCitation(newCitation);
 
 						// copy the citation's citationCollectionOrder
 						CitationCollectionOrder newCitationCollectionOrder = citationCollectionOrder.copy(this.getId(), newCitation.getId());
-						this.saveCitationCollectionOrder(newCitationCollectionOrder);
+						this.add(newCitationCollectionOrder);
+						if (isTemporary) {
```

Comment:
This logic looks to be the inverse of the commit message. You're only saving the citationcollectionorder if it is temporary.